### ISSUE TITLE
Unclear testing procedures

### ISSUE THREAD

**Issue by SiccarPoint** [landlab/landlab#758]
Per #752, the script `test-installed-landlab.py` has been removed. This means I need to change my dev workflow, since I was running this script to activate the tests locally. I'm pretty sure others in our team have been doing it this way too (@nicgaspar ?)

This isn't necessarily a problem, but it's pretty urgent that the relevant docs get updated with new best practice (which I now don't know!!):

https://landlab.readthedocs.io/en/master/dev_guide_install.html#testing

**Comment by nicgaspar** [comment]
@SiccarPoint You are right - I was definitely using the old way. The new way seems to be `pytest`. It seems to be working for me, and actually looks quite nice when it's run.

I think @margauxmouchene may be working on updating the docs?

**Comment by nicgaspar** [comment]
Literally on the command line just type `pytest`.
Note I was in my top landlab directory `nicgaspar\landlab`

Actually, it doesn't look like anyone in issues has talked about updating the wiki, etc. I can if we are full in now with `pytest`. @mcflugen ?

**Comment by mcflugen** [comment]
@SiccarPoint, from the command line, just `pytest`. If you don't have pytest installed,
```bash
$ conda install pytest
```
I thought we had updated the docs but I see now that we missed a spot.

**Comment by kbarnhart** [comment]
just typing `pytest` at the command prompt does all the defaults we want because @mcflugen put them in the `setup.cfg` file.

**Comment by SiccarPoint** [comment]
Cool, thanks! i.e., it behaves very much like `nose`, right? (which I had always avoided because it was really opaque to me how it worked…)

Does the Python version still work? i.e.,
```
import landlab
landlab.test()
```

This feels like a "cleaner", more intuitive way for beginner users to do it if they need to, as its still within Python and doesn't require command prompt stuff.

**Comment by mcflugen** [comment]
@SiccarPoint,

Yes, it behaves very much like `nose`.

No `landlab.test()` doesn't work anymore. We could definitely add something like that back in. Because we had decided we didn't really want users running the tests, I didn't worry about adding it back in. It would be pretty easy to add it back, though.

**Comment by SiccarPoint** [comment]
OK, if that's a conscious decision I can get behind it! As long as the documentation is there and clear, we're all good.

**Comment by margauxmouchene** [comment]
In the wiki, the 'test' we currently suggest is to check the version (for both developer and non-developer installs):
` > import landlab`
 `> landlab.__version__`
which is probably enough for users.

We can mention the pytest command in the Troubleshooting page.

There is a mention of the old test `landlab.test()` in README.rst and on [readthedocs](https://landlab.readthedocs.io/en/master/dev_guide_install.html#testing-the-landlab-installation). It looks like both of these have not been updated in a long time anyway. I can update that.
By the way, did we not decide to scrap the whole [Contributing to Landlab](http://landlab.readthedocs.io/en/latest/#contributing-to-landlab) section from readthedocs? It is a duplicate from what's on the wiki and, as we see here, we sometimes forget to update it along with the wiki. Readthedocs should be only code documentation.
